sale_purchase_force_vendor/models/sale_order_line.py

```diff
@@ -27,15 +27,23 @@ class SaleOrderLine(models.Model):
     @api.depends("product_id")
     def _compute_allowed_vendor_ids(self):
         supplierinfo_model = self.env["product.supplierinfo"]
+        partner_model = self.env["res.partner"]
         self.allowed_vendor_ids = self.allowed_vendor_ids
-        for item in self.filtered(lambda x: x.product_id):
+        for item in self.filtered(
+            lambda x: x.product_id and x.order_id.sale_purchase_force_vendor_restrict
```

Comment:
If you are here filtering section and notes, you should filter the proper field (`display_type`?)